sphinx_rtd_theme/versions.html

```diff
@@ -1,34 +1,136 @@
+      return "";
+    }
+    const versionsHTML = `
       <dl>
         <dt>{{ _('Versions') }}</dt>
-        {% for slug, url in versions %}
-          <dd><a href="{{ url }}">{{ slug }}</a></dd>
-        {% endfor %}
+        ${ config.versions.active.map(
+        (version) => `
+        <dd ${ version.slug === config.versions.current.slug ? 'class="rtd-current-item"' : '' }>
+          <a href="${ version.urls.documentation }">${ version.slug }</a>
+        </dd>
+        `).join("\n")}
       </dl>
+    `;
+    return versionsHTML;
+  }
+
+  function renderDownloads(config) {
+    if (!Object.keys(config.versions.current.downloads).length) {
+      return "";
+    }
+    const downloadsNameDisplay = {
+      pdf: "PDF",
+      epub: "Epub",
+      htmlzip: "HTML",
+    };
+
+    const downloadsHTML = `
       <dl>
         <dt>{{ _('Downloads') }}</dt>
-        {% for type, url in downloads %}
-          <dd><a href="{{ url }}">{{ type }}</a></dd>
-        {% endfor %}
-      </dl>
-      <dl>
-        {# Translators: The phrase "Read the Docs" is not translated #}
-        <dt>{{ _('On Read the Docs') }}</dt>
-          <dd>
-            <a href="//{{ PRODUCTION_DOMAIN }}/projects/{{ slug }}/?fromdocs={{ slug }}">{{ _('Project Home') }}</a>
-          </dd>
+        ${ Object.entries(config.versions.current.downloads).map(
+        ([name, url]) => `
           <dd>
-            <a href="//{{ PRODUCTION_DOMAIN }}/builds/{{ slug }}/?fromdocs={{ slug }}">{{ _('Builds') }}</a>
+            <a href="${ url }">${ downloadsNameDisplay[name] }</a>
           </dd>
+        `).join("\n")}
       </dl>
-    </div>
-  </div>
-{% endif %}
+    `;
+    return downloadsHTML;
+  }
+
+  document.addEventListener("readthedocs-addons-data-ready", function(event) {
+    const config = event.detail.data();
+
+    const flyout = `
+      <div class="rst-versions" data-toggle="rst-versions" role="note">
+        <span class="rst-current-version" data-toggle="rst-current-version">
+          <span class="fa fa-book"> Read the Docs</span>
+          v: ${ config.versions.current.slug }
+          <span class="fa fa-caret-down"></span>
+        </span>
+        <div class="rst-other-versions">
+          <div class="injected">
+            ${ renderLanguages(config) }
+            ${ renderVersions(config) }
+            ${ renderDownloads(config) }
+            <dl>
+              <dt>{{ _('On Read the Docs') }}</dt>
+              <dd>
+                <a href="${ config.projects.current.urls.home }">{{ _('Project Home') }}</a>
+              </dd>
+              <dd>
+                <a href="${ config.projects.current.urls.builds }">{{ _('Builds') }}</a>
+              </dd>
+              <dd>
+                <a href="${ config.projects.current.urls.downloads }">{{ _('Downloads') }}</a>
+              </dd>
+            </dl>
+            <dl>
+              <dt>{{ _('Search') }}</dt>
+              <dd>
+                <div style="padding: 6px;">
```

Comment:
This should be moved to our CSS rules, hardcoded element styles should be avoided.

Reply:
d16d19a